pkg/app/server.go

```diff
@@ -187,7 +187,7 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options) error {
 		// enable custom resource only
 		klog.InfoS("Used CRD resources only", "resources", resources)
 	default:
-		resources = append(resources, opts.Resources.AsSlice()...)
+		resources = opts.Resources.AsSlice()
```

Comment:
`default` case is when a `custom-resource-state-only` flag is not specified, and the `resources` flag is non-nil. In this case if we also specify `custom-resource-state-config` flag (with a CRS configuration), we'd get duplicate metrics if the entries in both overlap (for a common native resource).

`resources` in the code snippet above are the extracted resources we obtain from the CRS configuration, and removing them here would result in removing the current ability for KSM to generate and serve native and custom metrics (for different objects), for instance, `kube-state-metrics --kubeconfig=$KUBECONFIG --custom-resource-state-config-file=... --resources=...`.

IIUC with this change, the CRS resources extracted will be ignored entirely (for the aforementioned case), which will be a [breaking change](https://github.com/kubernetes/kube-state-metrics/blob/0f483430c406847ebafff91733e18fbb64f87f8e/pkg/app/server.go#L95C38-L97), since that's the expected behaviour ever since CRS was introduced.